app/models/shelter.rb

```diff
@@ -2,4 +2,12 @@ class Shelter < ApplicationRecord
   has_many :pets, dependent: :destroy
   has_many :reviews, dependent: :destroy
   validates_presence_of :name, :address, :city, :state, :zip
+
+  def self.three_highest_rated
+    Shelter.joins(:reviews)
+           .select('shelters.id, shelters.name, shelters.address, shelters.city, shelters.state, shelters.zip, avg(reviews.rating)')
```

Comment:
interested to know why we needed select here, since it looks like you are using all of the attributes

Reply:
It's because you also need to add another column to that table at the end, which is the averages. So you have to be explicit

Reply:
ooo i see, didnt even notice that last column listed, that makes sense.